src/NuGet.Core/NuGet.Configuration/Settings/Items/FileClientCertItem.cs

```diff
@@ -0,0 +1,233 @@
+            }
+
+            return element;
+        }
+
+        public override SettingBase Clone()
+        {
+            return new FileClientCertItem(PackageSource, FilePath, Password, IsPasswordIsClearText, Origin);
+        }
+
+        public override IEnumerable<X509Certificate> Search()
+        {
+            byte[] certificateData;
+
+            var filePath = FindAbsoluteFilePath();
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                throw new NuGetConfigurationException(string.Format(CultureInfo.CurrentCulture,
+                                                                    Resources.FileCertItemPathFileNotExist));
+            }
+
+            //Read certificate from file
+            using (FileStream stream = File.OpenRead(filePath))
+            {
+                certificateData = ReadStream(stream);
+            }
+
+            return new[] { string.IsNullOrWhiteSpace(Password) ? new X509Certificate2(certificateData) : new X509Certificate2(certificateData, Password) };
+        }
+
+        public void Update(string filePath, string password, bool storePasswordInClearText)
+        {
+            if (!string.IsNullOrWhiteSpace(filePath))
+            {
+                AddOrUpdateAttribute(ConfigurationConstants.PathAttribute, filePath);
+            }
+
+            if (!string.IsNullOrWhiteSpace(password))
+            {
+                if (storePasswordInClearText)
+                {
+                    AddOrUpdateAttribute(ConfigurationConstants.ClearTextPasswordAttribute, password);
+                    UpdateAttribute(ConfigurationConstants.PasswordAttribute, null);
+                }
+                else
+                {
+                    var encryptedPassword = EncryptionUtility.EncryptString(password);
+                    AddOrUpdateAttribute(ConfigurationConstants.PasswordAttribute, encryptedPassword);
+                    UpdateAttribute(ConfigurationConstants.ClearTextPasswordAttribute, null);
+                }
+            }
+        }
+
+        private string FindAbsoluteFilePath()
+        {
+            var originalValue = FilePath;
+            var expectedFiles = new List<string>
+            {
+                originalValue //Means absolute path
+            };
+
+            if (PathValidator.IsValidRelativePath(originalValue))
+            {
+                if (Origin != null)
+                {
+                    //Relative to config file path
+                    expectedFiles.Add(PathUtility.GetAbsolutePath(PathUtility.EnsureTrailingSlash(Origin.DirectoryPath), originalValue));
+                }
+
+                //Relative to current directory file path
+                expectedFiles.Add(PathUtility.GetAbsolutePath(PathUtility.EnsureTrailingSlash(Directory.GetCurrentDirectory()), originalValue));
+            }
+
+            return expectedFiles.FirstOrDefault(File.Exists);
+        }
```

Comment:
1. While putting paths to check in a list and then using LINQ to find the first that exists is a neat programming trick, I think it's less simple to read than using more basic procedural code and it avoid unnecessary allocations, particularly when the original value already exists.

2. The only other time I can think of a file or path being used in a `nuget.config` is when a local source is added. In that case, the path is always either absolute or relative to the `nuget.config`'s directory, never the current directory. So, code should already exist in this project to convert a relative path into an absolute path. Also, I don't support adding new semantics to the nuget.config file (search the process' current directory)

Reply:
1. Logic simplified. 
2. Code for "relative to absolute path transform" was written with your [PathUtility](https://github.com/NuGet/NuGet.Client/blob/d9a35fdd6a5fd39876358ec8b759766d112d0592/src/NuGet.Core/NuGet.Common/PathUtil/PathUtility.cs) and [PathValidator](https://github.com/NuGet/NuGet.Client/blob/4ec73f8165da896f44467a623c37a52a5e985a36/src/NuGet.Core/NuGet.Common/PathUtil/PathValidator.cs) classes. Current process directory analyze was removed.